src/box/memtx_engine.cc

```diff
@@ -74,6 +75,30 @@ template <class ALLOC>
 static inline void
 create_memtx_tuple_format_vtab(struct tuple_format_vtab *vtab);
 
+template <class ALLOC>
+static inline struct tuple *
+memtx_tuple_new_common_impl(struct tuple_format *format, const char *data,
+			    const char *end);
+
+template <class ALLOC>
+static inline struct tuple *
+memtx_tuple_compress_impl(struct tuple *tuple);
+
+struct tuple *
+(*memtx_tuple_new_common)(struct tuple_format *format, const char *data,
+			  const char *end);
+
+struct tuple *
+(*memtx_tuple_compress)(struct tuple *tuple);
```

Comment:
Remove memtx_tuple_new_common it's not necessary. We can't use format->vtab.tuple_new in memtx_tuple_compress, because this function(format->vtab.tuple_new) check field_map, but we should not do it for compressed data.